### PR TITLE
Split out completion from HieExtras

### LINKED ISSUE #1340: Split out completion logic
As per https://github.com/haskell/haskell-ide-engine/pull/1334 , the completion logic in HieExtras is getting pretty big. This should be split out into its own module

### PR DESCRIPTION
**Pull request by bubba**
Closes #1340

### CODE REVIEW

**Comment by alanz** [comment]
If this is just a rearrangement of code I am fine with it.

File: src/Haskell/Ide/Engine/Transport/LspStdio.hs [hunk 1]
```diff
@@ -41,6 +41,7 @@ import qualified GhcModCore               as GM ( loadMappedFileSource, getMMapp
 import           Haskell.Ide.Engine.Config
 import qualified Haskell.Ide.Engine.Ghc   as HIE
 import           Haskell.Ide.Engine.LSP.CodeActions
+import           Haskell.Ide.Engine.LSP.Completions
```
Comment (lorenzo) [inline src/Haskell/Ide/Engine/Transport/LspStdio.hs]: It would be nice if you imported this qualified. I personally find it easier to read when using the functions

Comment (alanz) [reply src/Haskell/Ide/Engine/Transport/LspStdio.hs]: @lorenzo this should be *only* a code move.  If we want to change the code that moved, that should be a separate PR.

Otherwise it is too hard to follow the trail of what happened. In my opinion.

**Comment by bubba** [comment]
Argh appveyor failed because of a chocolatey error. I'm going to ignore this one

File: src/Haskell/Ide/Engine/Transport/LspStdio.hs (hunk 1 above)
Comment (lorenzo) [reply src/Haskell/Ide/Engine/Transport/LspStdio.hs]: Well, the code was being used qualified before this move, so I think keeping the style is desirable.

Comment (bubba) [reply src/Haskell/Ide/Engine/Transport/LspStdio.hs]: Added it back!
